SculkCatalyst bloom API

Nacioszeczek opened the pull request:
Resolves PaperMC/Paper#9198

Nacioszeczek committed:
SculkCatalyst bloom API

Owen1212055 commented on patches/api/0421-SculkCatalyst-bloom-API.patch:
```diff
@@ -0,0 +1,23 @@
++     * @param location location to bloom at
++     * @param charge charge to bloom with, normally the amount of experience dropped from the dead entity
++     */
++    void bloom(org.bukkit.Location location, int charge);
```
Missing paper comments here

Owen1212055 commented on patches/server/0984-SculkCatalyst-bloom-API.patch:
```diff
@@ -0,0 +1,35 @@
++
++    // Paper start - SculkCatalyst bloom API
++    @Override
++    public void bloom(org.bukkit.Location location, int charge) {
```
In general, you can use the Position type instead of a location. This is because we don't want to have a world be a possible parameter, as it will always use the sculk's world anyway.

Nacioszeczek committed:
SculkCatalyst bloom API

Nacioszeczek committed:
restore AT

Machine-Maker commented on patches/server/0984-SculkCatalyst-bloom-API.patch:
```diff
@@ -0,0 +1,35 @@
++            world.getHandle().getRandom()
++        );
++        getTileEntity().getListener().getSculkSpreader().addCursors(
++            new net.minecraft.core.BlockPos(position.blockX(), (int) (position.blockY() + 0.5D), position.blockZ()),
```
Should we be doing this position mutation with the API method? I feel like its to offset from the emitter position within the game, but the API can specify whatever position it wants.

Nacioszeczek replied:
The +0.5 is copied directly from game's code. I assume it's for cases like the entity standing on e.g. a slab, if someone just passes an entity's Position imho should still add that half.

Unless I don't know what the emitterPos is

Nacioszeczek replied:
~~Or we could just have an 'overload' for LivingEntity~~ Even if someone specifies a block position that 0.5 ~~won't change anything since it's cast back to int~~ (nvm negative values), for an entity's position it may change things

Nacioszeczek committed:
SculkCatalyst bloom API

== AT ==
public net.minecraft.world.level.block.entity.SculkCatalystBlockEntity$CatalystListener bloom(Lnet/minecraft/server/level/ServerLevel;Lnet/minecraft/core/BlockPos;Lnet/minecraft/world/level/block/state/BlockState;Lnet/minecraft/util/RandomSource;)V
